tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2807,6 +2807,79 @@ def visit_import(self, node):
             pass
 
 
+class InvalidUseOfOverload(BaseChecker):
+    """Rule to check that use of the @overload decorator matches the async/sync nature of the underlying function"""
+
+    name = "invalid-use-of-overload"
+    priority = -1
+    msgs = {
+        "C4765": (
+            "Do not mix async and synchronous overloads",
+            "invalid-use-of-overload",
+            "Functions and their overloads must be either all async or all synchronous.",
+        ),
+    }
+
+    def visit_module(self, node):
+        """Check that use of the @overload decorator matches the async/sync nature of the underlying function"""
+        try:
+            klass = node.parent.parent.parent
```

Comment:
Are you trying to access the class here --  Could you maybe use visit_classdef and visit_functiondef? and track the functions/function names with a list that way?

Reply:
I have removed some code that wasn't needed anymore, I originally had it looking at a call level.

Do you want me to check at a module level if their are no classes or just change it so that I looks at a class level only?

Reply:
I just had a look and there are almost 53,000 occurances of @overload, I checked a few and they are all in a class.

Reply:
Looks good to me :)